src/deploy.sh

```diff
@@ -821,49 +821,103 @@ function modules_install_to()
   cmd_manager "$flag" "$cmd"
 }
 
-# This function handles the config file copy from the host machine to the
-# target.
+# This function is responsible for putting all the required boot files in a
+# single place (~/.cache/kw/to_deploy) to be deployed to the /boot folder
+# later. This function checks if there are dtb/dtbo files; if so, it moves
+# those files.
 #
-# @name Kernel name used in the deploy.
 # @flag How to display a command, the default value is
 #   "SILENT". For more options see `src/kwlib.sh` function `cmd_manager`
 # @target Target can be 1 (VM_TARGET), 2 (LOCAL_TARGET), and 3 (REMOTE_TARGET)
-# @build_and_deploy If the user uses `kw bd` we can safely copy the local
-#                   .config file.
-function deploy_config_file()
+# @kernel_img_name Kernel image file name, e.g., bzImage or Image.
+# @name Kernel name used during the deploy
+# @arch Target device architecture
+function pack_kernel_files_and_send()
 {
-  local name="$1"
-  local flag="$2"
-  local target="$3"
-  local build_and_deploy="$4"
+  local flag="$1"
+  local target="$2"
+  local kernel_img_name="$3"
+  local name="$4"
+  local arch=${5-${configurations[arch]}}
+  local build_and_deploy="$6"
+  local base_boot_path
+  local config_path
   local config_local_version
-  local complain_msg='Undefined .config file for the target kernel. Consider using kw bd'
-  local config_path="$PWD/.config"
-
+  local cache_boot_files_path
+  local compression_type
+  local cache_to_deploy_path
+  local dts_base_path
+  local tar_file_path
+  local kernel_name_arch
+
+  config_path='.config'
+  compression_type="${configurations[deploy_default_compression]}"
+  cache_to_deploy_path="${KW_CACHE_DIR}/${LOCAL_TO_DEPLOY_DIR}"
+  cache_boot_files_path="${cache_to_deploy_path}/boot"
+  tar_file_path="${cache_to_deploy_path}/${name}_boot.tar"
+
+  base_boot_path="arch/$arch_target/boot"
+  dts_base_path="${base_boot_path}/dts"
+
+  # Centralizing kernel files in a single place
+  mkdir -p "$cache_boot_files_path"
+
+  # 1. Copying .config file, we don't want to mislead developers by deploying
+  # the wrong config file.
   if [[ ! -f "$config_path" ]]; then
-    warning "$complain_msg"
-    return
+    warning 'Undefined .config file for the target kernel. Consider using kw bd'
```

Comment:
That is a long topic, but in summary, if we just use `kw d` we can copy the `.config` file, but we are not sure if the current build corresponds to the current config file. Think about this scenario:
1. You build the kernel
2. You replace the config file
3. You deploy your kernel

Notice that we are deploying a kernel in which the config file does not match the build kernel. However, we don't have this problem with kw bd because right after build the kernel; we deploy it, which does not give time for users to change the config file.

Reply:
Hmm.. got it. But I started from the point that I don't have any `.config` file in the folder and, therefore, this message showed up. From the situation you explained, the statement makes sense. However, I'm unsure if users can figure out what this guidance means if they don't handle any `.config` file.